.github/workflows/benchmark-action.yml

```diff
@@ -0,0 +1,112 @@
+name: Run performance benchmarks
+on:
+  workflow_dispatch: # Allows manual triggering of workflow
+  push:
+    branches:
+      - main
+    paths:
+      - src/client/Microsoft.Identity.Client/**/*.cs
+  #pull_request: # Will re-enable once the code to not push to gh-pages is fixed
```

Comment:
I would think 2 more scenarios are relevant, as a follow-up

1. Creation of CCA object 
2. Using cache serialization

Reply:
1. We already have a test for that, just not running it for automation. Just need to add.
2. We already have that.